Sources/SnapshottingTests/PreviewLayoutTest.swift

```diff
@@ -10,25 +10,46 @@ import Foundation
 import SwiftUI
 import XCTest
 
-// Test Xcode previews by forcing a layout pass of each one
+/// A test class for verifying Xcode previews by forcing a layout pass on each one.
+///
+/// This class is designed to discover and test SwiftUI previews to ensure they can be laid out without crashing.
+/// It provides mechanisms for filtering previews and performs layout tests on different platforms.
 open class PreviewLayoutTest: PreviewBaseTest, PreviewFilters {
 
+  /// Returns an optional array of preview names to be included in the snapshot testing. This also supports Regex format.
+  ///
+  /// Override this method to specify which previews should be included in the test.
+  /// - Returns: An optional array of String containing the names of previews to be included.
   open class func snapshotPreviews() -> [String]?  {
     nil
   }
 
+  /// Returns an optional array of preview names to be excluded from the snapshot testing. This also supports Regex format
+  ///
+  /// Override this method to specify which previews should be excluded from the test.
+  /// - Returns: An optional array of String containing the names of previews to be excluded.
   open class func excludedSnapshotPreviews() -> [String]? {
     nil
   }
 
   static private var previews: [PreviewType] = []
 
+  /// Discovers all relevant previews based on inclusion and exclusion filters.
+  ///
+  /// This method uses `FindPreviews` to locate all previews, applying any specified filters.
+  /// - Returns: An array of `DiscoveredPreview` objects representing the found previews.
```

Comment:
Perhaps mention that subclasses should *not* override this method, same for `testPreview` and same for the SnapshotTest class